### DISCUSSION TITLE
Certificates are not Generating

### DISCUSSION THREAD

**Discussion by Venkatesh-Dayani23** [Sunbird-RC/community#1004] [Issues]
Hi Team,
Recently we have upgraded to Sunbird 5.1.0 after the upgrade the Certificates are not generating. Below are the error logs.

**certificatesign error logs**

Credential template loaded from cache                                                                                               │
│ SyntaxError: Unexpected token s in JSON at position 0                                                                               │
│     at JSON.parse (<anonymous>)                                                                                                     │
│     at generateCredentials (/app/src/services/signerService.js:41:33)                                                               │
│     at generateCredentialsRoute (/app/src/routes/signerController.js:27:18)                                                         │
│     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)                                                   │
│     at async Server.<anonymous> (/app/main.js:17:32)  

**registry service logs**

Caused by: dev.sunbirdrc.registry.exception.SignatureException$UnreachableException: Unable to reach service: 500 Internal Server E │
│ rror: [{"error": "Internal Server Error", "message": Unexpected token s in JSON at position 0}]                                     │
│     at dev.sunbirdrc.registry.service.impl.SignatureServiceImpl.sign(SignatureServiceImpl.java:96)                                  │
│     at dev.sunbirdrc.registry.service.impl.RegistryServiceImpl.generateCredentials(RegistryServiceImpl.java:253)                    │
│     at dev.sunbirdrc.registry.service.impl.RegistryServiceImpl.addEntity(RegistryServiceImpl.java:197)                              │
│     at sun.reflect.GeneratedMethodAccessor80.invoke(Unknown Source)                                                                 │
│     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)                                        │
│     at java.lang.reflect.Method.invoke(Method.java:498)                                                                             │
│     at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)                                   │
│     at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:205)                                     │
│     at com.sun.proxy.$Proxy97.addEntity(Unknown Source)                                                                             │
│     at dev.sunbirdrc.registry.helper.RegistryHelper.addEntity(RegistryHelper.java:253)                                              │
│     ... 66 common frames omitted      

cc: @raghupathiguduri

**Comment by Venkatesh-Dayani23** [comment]
@vinukumar-vs Can you help on this issue ?

**Reply by Venkatesh-Dayani23** [reply]
@amitpriyadarshi
Can you help on this ?

**Comment by Mothukuri-venu** [comment]
Hi Venkatesh,
The issue looks like it is coming from the **certificate-sign service while parsing the credential template**.
The key error is:
```
SyntaxError: Unexpected token s in JSON at position 0
at JSON.parse()
at generateCredentials (/app/src/services/signerService.js:41:33)
```
This means the signer service is expecting a valid JSON credential template, but it is receiving invalid JSON content. Since the error starts after loading the template from cache:
```
Credential template loaded from cache
```
the possible causes are:
1. **Invalid credential template stored in cache**
   * The cached template may be corrupted or stored as a string that is not valid JSON.
   * Try clearing the credential template cache and restart the certificate-sign service.
2. **Credential template format changed after upgrade**
   * After upgrading to Sunbird RC 5.1.0, verify that all credential templates are compatible with the new version.
   * Re-import/update the templates if they were created in an older version.
3. **Check the template content**
   * The character `s` at JSON position 0 usually means the service received something like:
     ```
     schema...
     ```
     or another plain string instead of:
     ```json
     {
       "type": "...",
       "credentialSubject": {}
     }
     ```
   * Please check the exact response/template returned from the registry or cache.
4. **Verify configuration between services**
   * Ensure the `certificate-sign` service URL configured in the registry service is pointing to the correct upgraded service.
   * Confirm both registry and certificate-sign services are running on compatible versions.
Suggested troubleshooting steps:
```bash
# Restart certificate-sign service after clearing cache
docker restart <certificate-sign-container>
# Check service logs
docker logs -f <certificate-sign-container>
```
If the issue continues, please share:
* The credential template JSON being used.
* Whether the upgrade was done with database migration scripts.
* The deployment method (Docker/Kubernetes).
* The certificate-sign service version.
The registry error is only a downstream failure; the actual root cause is the invalid JSON received by the signer service.
